test/integration/test_integration_basics.py

```diff
@@ -52,37 +55,60 @@ def mini_stress_test_h1(fixture, args):
   return counters
 
 
+# The mini stress tests below are executing in closed-loop mode. As we guard the pool against
+# overflows, we can set fixed expectations with respect to overflows and anticipated pending
+# totals.
 def test_http_h1_mini_stress_test_with_client_side_queueing(http_test_server_fixture):
   """
   Run a max rps test with the h1 pool against our test server, using a small client-side
-  queue. We expect to observe:
-  - upstream_rq_pending_total increasing
-  - upstream_cx_overflow overflows
-  - blocking to be reported by the sequencer
-  """
-  counters = mini_stress_test_h1(http_test_server_fixture, [
+  queue."""
+  counters = mini_stress_test(http_test_server_fixture, [
       http_test_server_fixture.getTestServerRootUri(), "--rps", "999999", "--max-pending-requests",
-      "10", "--duration 10"
+      "10", "--duration 10", "--connections", "1"
   ])
-  assertCounterGreater(counters, "upstream_rq_pending_total", 100)
-  assertCounterGreater(counters, "upstream_cx_overflow", 0)
+  assertCounterEqual(counters, "upstream_rq_pending_total", 11)
+  assertCounterEqual(counters, "upstream_cx_overflow", 10)
 
 
 def test_http_h1_mini_stress_test_without_client_side_queueing(http_test_server_fixture):
   """
   Run a max rps test with the h1 pool against our test server, with no client-side
-  queueing. We expect to observe:
-  - upstream_rq_pending_total to be equal to 1
-  - blocking to be reported by the sequencer
-  - no upstream_cx_overflows
+  queueing.
   """
-  counters = mini_stress_test_h1(
-      http_test_server_fixture,
-      [http_test_server_fixture.getTestServerRootUri(), "--rps", "999999", "--duration 2"])
+  counters = mini_stress_test(http_test_server_fixture, [
+      http_test_server_fixture.getTestServerRootUri(), "--rps", "999999", "--duration 2",
+      "--connections", "1"
+  ])
   assertCounterEqual(counters, "upstream_rq_pending_total", 1)
   assertNotIn("upstream_cx_overflow", counters)
 
 
+def test_http_h2_mini_stress_test_with_client_side_queueing(http_test_server_fixture):
+  """
+  Run a max rps test with the h2 pool against our test server, using a small client-side
+  queue. 
+  """
+  counters = mini_stress_test(http_test_server_fixture, [
+      http_test_server_fixture.getTestServerRootUri(), "--rps", "999999", "--max-pending-requests",
+      "10", "--duration 10", "--h2", "--max-active-requests", "1", "--connections", "1"
+  ])
+  assertCounterEqual(counters, "upstream_rq_pending_total", 1)
+  assertCounterEqual(counters, "upstream_rq_pending_overflow", 10)
+
+
+def test_http_h2_mini_stress_test_without_client_side_queueing(http_test_server_fixture):
+  """
+  Run a max rps test with the h2 pool against our test server, with no client-side
+  queueing. 
+  """
+  counters = mini_stress_test(http_test_server_fixture, [
+      http_test_server_fixture.getTestServerRootUri(), "--rps", "999999", "--duration 2", "--h2",
```

Comment:
Should all these duration args be
`"--duration", "2", `
?

Reply:
yes, nice catch. It won't break us here, but sticking to this convention allows the `mini_stress_test()` to easily query which flags are set.

Reply:
Fixed in 467ff20ed1387c83d044f8931bee454c4a172a39